adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -558,6 +599,433 @@ public void testVerifyBrokerRedirectUriInvalidSignature() throws PackageManager.
+        } finally {
+            cacheStore.removeAll();
+        }
+    }
+
+    /**
+     * Test for Resiliency Feature when the AT is not expired and outage mode is on
+     *
+     * If
+     * 1. outage mode is on, which means ExtendedLifetimeEnabled is set to be true in AuthenticationContext
+     * 2. the token cache item is not expired and also not expired in the extended expire time
+     * 3. the server is not down where we get the 500, 503, 504 error at the first try and succeed retry
+     * Then
+     * The authentication result with new access token should be returned.
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_outageModeOn_validAT() throws PackageManager.NameNotFoundException,
+            NoSuchAlgorithmException, OperationCanceledException, IOException, AuthenticatorException,
+            InterruptedException {
+        // make sure AT's expires_in is expired and ext_expires_in is not expired
+        final Calendar expiredTime = new GregorianCalendar();
+        expiredTime.add(Calendar.MINUTE, -MINUS_MINUITE);
+        final Calendar extendedExpiresTime = new GregorianCalendar();
+        extendedExpiresTime.add(Calendar.MINUTE, EXTEND_MINUS_MINUTE);
+        final ITokenCacheStore cacheStore = getExtendedLifetimeTokenCache(expiredTime.getTime(), false, false, extendedExpiresTime.getTime());
+
+        final AccountManager mockedAccountManager = getMockedAccountManager();
+        mockAccountManagerGetAccountBehavior(mockedAccountManager);
+        final FileMockContext mockContext = createMockContext();
+        mockContext.setMockedAccountManager(mockedAccountManager);
+        //set useBroker into false
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
+
+        //mock http response
+        final HttpURLConnection mockedConnection = Mockito.mock(HttpURLConnection.class);
+        HttpUrlConnectionFactory.mockedConnection = mockedConnection;
+        Util.prepareMockedUrlConnection(mockedConnection);
+        Mockito.when(mockedConnection.getOutputStream()).thenReturn(Mockito.mock(OutputStream.class));
+        Mockito.when(mockedConnection.getInputStream()).thenReturn(Util.createInputStream(Util.getSuccessTokenResponse(false, false)),
+                Util.createInputStream(Util.getSuccessTokenResponse(true, true)));
+        Mockito.when(mockedConnection.getResponseCode()).thenReturn(HttpURLConnection.HTTP_GATEWAY_TIMEOUT, HttpURLConnection.HTTP_OK);
+
+
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false, cacheStore);
+        //turn on the outage mode
+        authContext.setExtendedLifetimeEnabled(true);
+
+        final TestAuthCallback callback = new TestAuthCallback();
+        authContext.acquireTokenSilentAsync("resource", "clientid", TEST_USERID, callback);
+
+        final CountDownLatch signal = new CountDownLatch(1);
```

Comment:
use sync call, then no need to have the countdownlatch as well as the call back.